src/tensorwaves/optimizer/minuit.py

```diff
@@ -16,27 +16,30 @@ class Minuit2(Optimizer):
     Implements the `~.interfaces.Optimizer` interface.
     """
 
-    def __init__(self) -> None:
-        pass
+    def __init__(self, callbacks: Optional[Iterable[Callback]] = None) -> None:
```

Comment:
I like this `CallbackList` idea. Can't the callbacks argument here be reduced to `Optional[Callback]` and in case multiple callbacks are wanted, a `CallbackList` can be used as the argument. For that to work the `CallbackList` should implement the `Callback` interface though.

Reply:
The idea comes from
https://www.tensorflow.org/api_docs/python/tf/keras/callbacks/CallbackList
though the existing implementation is just to clean up the code in the `optimize` method a bit.

Reply:
Looks better now :smirk: 5dba9c4

Reply:
Yeah, but they also make the `CallbackList` implement the `Callback`  interface right?